Source/DafnyLanguageServer/Language/DafnyLangParser.cs

```diff
@@ -60,23 +60,23 @@ public sealed class DafnyLangParser : IDafnyParser, IDisposable {
             new(errorReporter.Options, document.Uri.ToUri(), document.Content)
           },
           errorReporter, cancellationToken);
+        cachingParser.Prune();
         return result;
       }
       finally {
         telemetryPublisher.PublishTime("Parse", document.Uri.ToString(), DateTime.Now - beforeParsing);
-        cachingParser.Prune();
```

Comment:
Why not prune the cache even in case of parse errors?

Reply:
`ParseFiles` doesn't throw an exception if there are parse errors, but it does do that if parsing was cancelled, which means some of the includes are likely not to be traversed which would change the effect of pruning, which will prunes any entry that wasn't hit since the last prune.